### ISSUE TITLE
ui: negative bytes values on node overview page

### ISSUE THREAD

**Issue by cuongdo** [cockroachdb/cockroach#5932]
Please answer these questions **for each node in your cluster**:

1. What version of CockroachDB are you using (`cockroach version`)?
bec91e4a845a1e430898305970eada242be4b2da

2. What operating system and processor architecture are you using?
Linux cockroach-0 4.2.0-16-generic #19-Ubuntu SMP Thu Oct 8 15:35:06 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

3. What flags/environment variables did you pass to `cockroach start`?
`./cockroach start --log-dir=cockroach-data/logs --logtostderr=false --insecure --host=10.142.0.3 --port=26257 --http-port=8080 --join=10.142.0.7`

Please describe the issue you observed:

1. What did you do?
Ran terraform on GCE with 5 nodes. Went to admin UI nodes page without putting any load on the cluster.

2. What did you expect to see?
Non-negative bytes values.

3. What did you see instead?
Negative bytes for a node.

<img width="531" alt="screenshot 2016-04-07 14 45 30" src="https://cloud.githubusercontent.com/assets/1564582/14363277/6e3482fc-fcd1-11e5-9297-48dd435ac533.png">

**Comment by cuongdo** [comment]
As requested, I reran with `--vmodule=replica_gc_queue=1`. I got a negative byte count on one node but no log messages from `replica_gc_queue.go`.

Here's the current store info from `/_status/metrics/local`:

```
"stores": {
    "5": {
      "capacity": 10365558784,
      "capacity.available": 9207152640,
      "gcbytesage": 0,
      "intentage": 0,
      "intentbytes": 0,
      "intentcount": 0,
      "keybytes": -5358,
      "keycount": -134,
      "lastupdatenanos": 1460056170759915446,
      "livebytes": -4901,
      "livecount": -134,
      "range.adds": 0,
      "range.removes": 0,
      "range.splits": 0,
      "ranges.available": 0,
      "ranges.leader": 0,
      "ranges.replicated": 0,
      "replicas": 2,
      "rocksdb.block.cache.hits": 0,
      "rocksdb.block.cache.misses": 0,
      "rocksdb.block.cache.pinned-usage": 0,
      "rocksdb.block.cache.usage": 0,
      "rocksdb.bloom.filter.prefix.checked": 0,
      "rocksdb.bloom.filter.prefix.useful": 0,
      "rocksdb.compactions": 0,
      "rocksdb.flushes": 0,
      "rocksdb.memtable.hits": 0,
      "rocksdb.memtable.misses": 0,
      "rocksdb.memtable.total-size": 44444,
      "rocksdb.table-readers-mem-estimate": 0,
      "sysbytes": 1487,
      "syscount": 14,
      "valbytes": 457,
      "valcount": -134
    }
```

**Comment by tschottdorf** [comment]
We should put in an assertion in `(*rangeStats).MergeMVCCStats` (or up from
the call so that we can see the request at which it explodes). I don't
think there should ever be any negative values in it at that point, and it
should catch any negative number you could see on the UI.

That doesn't tell you where it came from (just that something went wrong at
that command or anytime before), but it's a start.

On Thu, Apr 7, 2016 at 3:14 PM Cuong Do <notifications@github.com> wrote:

> As requested, I reran with --vmodule=replica_gc_queue=1. I got a negative
> byte count on one node but no log messages from replica_gc_queue.go.
>
> Here's the current store info from /_status/metrics/local:
>
> "stores": {
>     "5": {
>       "capacity": 10365558784,
>       "capacity.available": 9207152640,
>       "gcbytesage": 0,
>       "intentage": 0,
>       "intentbytes": 0,
>       "intentcount": 0,
>       "keybytes": -5358,
>       "keycount": -134,
>       "lastupdatenanos": 1460056170759915446,
>       "livebytes": -4901,
>       "livecount": -134,
>       "range.adds": 0,
>       "range.removes": 0,
>       "range.splits": 0,
>       "ranges.available": 0,
>       "ranges.leader": 0,
>       "ranges.replicated": 0,
>       "replicas": 2,
>       "rocksdb.block.cache.hits": 0,
>       "rocksdb.block.cache.misses": 0,
>       "rocksdb.block.cache.pinned-usage": 0,
>       "rocksdb.block.cache.usage": 0,
>       "rocksdb.bloom.filter.prefix.checked": 0,
>       "rocksdb.bloom.filter.prefix.useful": 0,
>       "rocksdb.compactions": 0,
>       "rocksdb.flushes": 0,
>       "rocksdb.memtable.hits": 0,
>       "rocksdb.memtable.misses": 0,
>       "rocksdb.memtable.total-size": 44444,
>       "rocksdb.table-readers-mem-estimate": 0,
>       "sysbytes": 1487,
>       "syscount": 14,
>       "valbytes": 457,
>       "valcount": -134
>     }
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/issues/5932#issuecomment-207054535>
>
-- 

-- Tobias

**Comment by tschottdorf** [comment]
See #7290
